### PR TITLE
Eliminate @storis dependencies internalize configs

### PR DESCRIPTION
**Pull request by philfuster**
This PR removes the two storis dependencies `@storis/eslint-config` and `@storis/tsconfig`. Those repositories are no longer being maintained, so their configs will be internalized.

This PR also updates the `eslint-config-prettier` plugin.

There was an attempt made to upgrade eslint to v9 as well, but there are plugins that are yet to support v9 of eslint. They have been documented in a tracker issue - [#646](https://github.com/STORIS/mvom/issues/646)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -21,11 +28,278 @@ module.exports = {
...
+			// do not enforce format on property names
+			{ selector: 'property', format: null },
+			// PascalCase for classes and TypeScript keywords
+			{
+				selector: ['typeLike'],
+				format: ['PascalCase'],
+			},
+			// allow trailing ASC and DESC on enumerations
+			{
+				selector: 'enumMember',
```
Comment (shawnmcknight) [inline .eslintrc.js:58]: This rule is likely not relevant for this repo

File: .eslintrc.js [hunk 2]
```diff
@@ -21,11 +28,278 @@ module.exports = {
...
+			// allow variables to be camelCase or UPPER_CASE
+			{ selector: 'variable', format: ['camelCase', 'UPPER_CASE'] },
+			// allow known variable exclusions
+			{
+				selector: 'variable',
+				filter: { regex: '^(_id|__v|_raw)$', match: true },
+				format: null,
+			},
+			// GraphQL variables PascalCase
+			{
```
Comment (shawnmcknight) [inline .eslintrc.js:44]: This rule is likely not relevant for this repo

File: .eslintrc.js [hunk 3]
```diff
@@ -21,11 +28,278 @@ module.exports = {
...
+		{
+			files: tsGlobs,
+			rules: {
+				// enforce return types on module boundaries
+				'@typescript-eslint/explicit-module-boundary-types': 'error',
+			},
+		},
+
+		{
+			files: ['**/*.types.ts', '**/types/*.ts', '**/*.schema.ts', '**/instances/**'],
```
Comment (shawnmcknight) [inline .eslintrc.js:75]: I think the `**/types/*.ts` file override is relevant for this repo, but I don't believe any of the others are.

File: .eslintrc.js [hunk 4]
```diff
@@ -21,11 +28,278 @@ module.exports = {
...
+				// https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
+				'@typescript-eslint/no-unused-vars': 'error',
+			},
+		},
+		{
+			files: tsGlobs,
+			extends: ['plugin:@typescript-eslint/recommended-requiring-type-checking'],
+			rules: {
+				// ban ts-comment except with description
+				'@typescript-eslint/ban-ts-comment': [
```
Comment (shawnmcknight) [inline .eslintrc.js:171]: I think we should make this:
```
				// ban ts-comment except with description
				'@typescript-eslint/ban-ts-comment': [
					'error',
					{
						'ts-expect-error': { descriptionFormat: '^: ' },
						'ts-ignore': true,
						'ts-nocheck': true,
						'ts-check': false,
					},
				],
```

File: .eslintrc.js [hunk 5]
```diff
@@ -21,11 +28,278 @@ module.exports = {
...
+						'ts-ignore': 'allow-with-description',
+						'ts-nocheck': false,
+						'ts-check': false,
+					},
+				],
+			},
+		},
+
+		{
+			files: ['**/index.[jt]s?(x)', '**/*.gql.[jt]s', '**/constants.ts'],
```
Comment (shawnmcknight) [inline .eslintrc.js:216]: The `gql` suffix is definitely not needed as an override. I don't think the `constants` one is needed.

File: tsconfig.json [hunk 6]
```diff
@@ -1,20 +1,118 @@
...
   },
   "include": [
     "src/**/*", // source folder
     "scripts/**/*", // scripts folder
     "test/**/*", // test helpers
     "./*.js", // root javascript files
     "./.*.js" // root javascript config files
-  ]
+  ],
+  "ts-node": {
```
Comment (shawnmcknight) [inline tsconfig.json:113]: This isn't relevant to this repo

**Review by shawnmcknight — CHANGES_REQUESTED** [review]
Overall looks good.  The biggest issues are some things that are not relevant to this repo that leaked into the configs.

File: .eslintrc.js (hunk 4 above)
Comment (philfuster) [reply .eslintrc.js:171]: Good idea. Looking at all these rules has been great. Thank you.

File: .eslintrc.js (hunk 1 above)
Comment (philfuster) [reply .eslintrc.js]: fixed.

File: .eslintrc.js (hunk 2 above)
Comment (philfuster) [reply .eslintrc.js]: fixed.

File: .eslintrc.js (hunk 3 above)
Comment (philfuster) [reply .eslintrc.js]: fixed.

File: .eslintrc.js (hunk 5 above)
Comment (philfuster) [reply .eslintrc.js]: removed `gql` and `constants` file matches.

File: tsconfig.json (hunk 6 above)
Comment (philfuster) [reply tsconfig.json]: done. removed ts-node config

**Review by shawnmcknight — APPROVED** [review]
LGTM!

**Pull request merged**